src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1731,14 +1955,24 @@ private static bool AreEqualityComparersEqual(HashSet<T> set1, HashSet<T> set2)
         /// Workaround Comparers that throw ArgumentNullException for GetHashCode(null).
         /// </summary>
         /// <param name="item"></param>
+        /// <param name="comparer"></param>
         /// <returns>hash code</returns>
-        private int InternalGetHashCode(T item)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int InternalGetHashCode(T item, IEqualityComparer<T>? comparer)
```

Comment:
Wrappers like this add overhead for larger structs; even with AggressiveInlining.

(similar comment as the other one you have fixed already)